### PR TITLE
DATAUP-317: ESLint / Prettier: running prettier over the codebase

### PR DESCRIPTION
**Pull request by ialarmedalien**
# Description of PR purpose/changes

Use standardised formatting for all kbase JS files, courtesy of `prettier`, with the aim of having all JS files formatted and linted using eslint and prettier.

This is part 1 of the process, formatting the files; in the next step, I will run `eslint --fix`, which can automatically fix some grammar issues and will highlight other problems.

Prettier has a 'debug check' mode, where it checks the abstract syntax tree (AST) of a file before and after code formatting. When I ran `prettier --debug-check '**/*.js'` over the repo, there were six files where formatting resulted in a change in the AST. I hence split up this PR into four commits:

* [Commit 1](https://github.com/kbase/narrative/pull/1979/commits/fc6a2935ca0ca4ac4512df170973f15e63368628): formats all files minus the six files with AST changes; see the `.prettierignore` file for the files that were not formatted.
* [Commit 2](https://github.com/kbase/narrative/pull/1979/commits/571e1dd0824f58711e7c7284ba3b7d94db122ad7): five files with small differences in AST; I've highlighted the places where the AST differs with comments in the commit.
* [Commit 3](https://github.com/kbase/narrative/pull/1979/commits/df0cfe2fb4638985f038bdcba15d62fd9028a04d): another run, with no JS files in the `.prettierignore` file.
* [Commit 4](https://github.com/kbase/narrative/pull/1979/commits/dd6cca238e8ef1d6c6961464478e058f353f3b33): fixing comment indentation on a file where prettier had left it not looking as pretty as it could.

See also https://github.com/kbase/narrative/pull/1978, which puts the npm machinery in place for these changes.

# Testing Instructions
* Details for how to test the PR: 
- [ ] Tests pass in travis and locally -- tests are currently broken.
- [x] Changes available by checking out the repo and seeing that the code is now formatted.

# Dev Checklist:

- [x] My code follows the guidelines at https://sites.google.com/truss.works/kbasetruss/development
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: kbase-extension/static/kbase/js/widgets/function_output/kbaseGenomeAnnotationViewer.js [hunk 1]
```diff
@@ -344,341 +350,397 @@ define (
...
         },
 
+        setDataTableStyles: function ($table) {
+            return $table.addClass('table table-striped table-bordered table-hover').css({
+                width: '100%',
+                border: '1px solid #ddd',
+                'margin-left': 'auto',
+                'margin-right': 'auto',
+            });
+        },
```
Comment (ialarmedalien) [inline kbase-extension/static/kbase/js/widgets/function_output/kbaseGenomeAnnotationViewer.js:408]: This is the only place in this file where prettier has introduced a change to the abstract syntax tree of the file. From a code perspective, there is no difference between these lines and lines 395-399 in the old file.

File: kbase-extension/static/kbase/js/widgets/narrative_core/catalog/kbaseCatalogBrowser.js [hunk 2]
```diff
@@ -437,81 +482,82 @@ define ([
...
-
-        initMainPanel: function($appListPanel, $moduleListPanel) {
-            var $mainPanel = $('<div>').addClass('container').css({'width':'100%','overflow-x':'hidden','height':'550px','overflow-y': 'auto'});
+        initMainPanel: function ($appListPanel, $moduleListPanel) {
+            var $mainPanel = $('<div>').addClass('container').css({
+                width: '100%',
+                'overflow-x': 'hidden',
+                height: '550px',
+                'overflow-y': 'auto',
+            });
```
Comment (ialarmedalien) [inline kbase-extension/static/kbase/js/widgets/narrative_core/catalog/kbaseCatalogBrowser.js:522]: AST change -- no functional change.

File: kbase-extension/static/kbase/js/widgets/narrative_core/kbaseNarrativeMethodCell.js [hunk 3]
```diff
@@ -637,152 +693,164 @@ define([
...
-                errorText = $('<div>');  // class="kb-jobs-error-modal">');
+                errorType = 'Runtime';
+                if (jobState.state.error === 'awe_error') errorType = 'AWE Error';
+            } else if (Object.keys(jobState.state.step_errors).length !== 0) {
+                /* error types:
+                 * 1. jobState.state.error is a real string. Just cough it up.
+                 * 2. jobState.state is missing
+                 * 3. jobInfo is partly missing (e.g., lost the cell that it should point to)
+                 * 4. jobInfo is still partly missing (e.g., dont' know what cell it should point to)
+                 */
```
Comment (ialarmedalien) [inline kbase-extension/static/kbase/js/widgets/narrative_core/kbaseNarrativeMethodCell.js:781]: AST change, no change in functionality

File: kbase-extension/static/kbase/js/widgets/vis/plants/kbasePMIBarchart.js [hunk 4]
```diff
@@ -228,86 +218,89 @@ define (
...
-                            )
-                        ;
-
-
+                                $.jqElem('ul').addClass('dropdown-menu').css({
+                                    width: '450px',
+                                    'padding-left': '5px',
+                                    'text-align': 'left',
+                                })
+                            );
```
Comment (ialarmedalien) [inline kbase-extension/static/kbase/js/widgets/vis/plants/kbasePMIBarchart.js:237]: AST change, no functional difference

File: kbase-extension/static/kbase/js/widgets/narrative_core/kbaseNarrativeOutputCell.js [hunk 5]
```diff
@@ -239,57 +230,56 @@ define([
...
+                require([widget], function (W) {
+                    if (self.$widgetBody) {
+                        self.$widgetBody.remove();
                     }
-                );
+                    self.$widgetBody = $('<div>');
+                    self.$body.append(self.$widgetBody);
+                    self.$outWidget = new W(self.$widgetBody, widgetData);
+                    resolve();
+                }, // otherwise on error the promise will dangle. // TODO: No, should reject the promise, or handle here and resolve,
```
Comment (ialarmedalien) [inline kbase-extension/static/kbase/js/widgets/narrative_core/kbaseNarrativeOutputCell.js]: The AST has been changed here. I've reformatted these comments in a later commit as they don't look good like this.

**Comment by bio-boris** [comment]
@ialarmedalien We are closing in favor of doing it right before we will deploy it so as to not have to work on merge conflicts

**Pull request closed without merging**